man/man4/zfs.4

```diff
@@ -487,7 +487,12 @@ However, this is limited by
 .It Sy zfetch_array_rd_sz Ns = Ns Sy 1048576 Ns B Po 1 MiB Pc Pq ulong
 If prefetching is enabled, disable prefetching for reads larger than this size.
 .
-.It Sy zfetch_max_distance Ns = Ns Sy 8388608 Ns B Po 8 MiB Pc Pq uint
+.It Sy zfetch_min_distance Ns = Ns Sy 4194304 Ns B Po 4 MiB Pc Pq uint
+Min bytes to prefetch per stream.
+Prefetch distance doubles each hit until it reaches this value.
+After that it grows by 1/8 each hit only until it cover the read latency.
```

Comment:
Can you describe what you mean here, so we can improve the wording: "grows by 1/8 each hit only until it cover the read latency" is grammatically awkward.

Should we mention that the prefetch starts with the demand access size, and doubles until it reaches `zfetch_min_distance`? (this might not be the best name, since it is not the minimum prefetch, but I think it is fine)

Reply:
Prefetch indeed starts from the demand access size (or at least a block size), same as before.  Then it very quickly reaches `zfetch_min_distance`, which extreme speed makes it practically a mandatory minimum.  After that on every speculative data read completion its offset is compared to the current demand access position, and, if it appears to be behind, the zs_more flag is set, seeing which next demand read will increase prefetch distance by 1/8.  Since speculative reads often complete in batches and the first completes together with demand read, effectively that distance increase by 1/8 does not happen on every request, but once in several megabytes or more, depending how close it is to satisfying the reader bandwidth.

Reply:
I've rewritten it more verbose.  Hope that's better.